cola/main.py

```diff
@@ -0,0 +1,480 @@
+    return application_start(context, view)
+
+
+def cmd_version(args):
+    from cola import version
+    version.print_version(brief=args.brief)
+    return 0
+
+
+def cmd_pull(args):
+    context = application_init(args, update=True)
+    from cola.widgets import remote
+    view = remote.pull()
+    if args.rebase:
+        view.set_rebase(True)
+    return application_start(context, view)
+
+
+def cmd_push(args):
+    context = application_init(args, update=True)
+    from cola.widgets import remote
+    view = remote.push()
+    return application_start(context, view)
+
+
+def cmd_rebase(args):
+    kwargs = {
+            'verbose': args.verbose,
+            'quiet': args.quiet,
+            'autostash': args.autostash,
+            'fork_point': args.fork_point,
+            'onto': args.onto,
+            'preserve_merges': args.preserve_merges,
+            'strategy': args.strategy,
+            'no_ff': args.no_ff,
+            'merge': args.merge,
+            'exec': getattr(args, 'exec', None), # python keyword
+            'keep_empty': args.keep_empty,
+            'force_rebase': args.force_rebase,
+            'strategy_option': args.strategy_option,
+            'stat': args.stat,
+            'no_stat': args.no_stat,
+            'verify': args.verify,
+            'rerere_autoupdate': args.rerere_autoupdate,
+            'root': args.root,
+            'autosquash': args.autosquash,
+            'committer_date_is_author_date': args.committer_date_is_author_date,
+            'ignore_date': args.ignore_date,
+            'whitespace': args.whitespace,
+            'ignore_whitespace': args.ignore_whitespace,
+            'C': args.context_lines,
+            'continue': getattr(args, 'continue', False), # python keyword
+            'abort': args.abort,
+            'skip': args.skip,
+            'edit_todo': args.edit_todo,
+            'upstream': args.upstream,
+            'branch': args.branch,
+            'capture_output': False,
+    }
+    status, out, err = cmds.do(cmds.Rebase, **kwargs)
+    if out:
+        core.stdout(out)
+    if err:
+        core.stderr(err)
+    return status
+
+
+def cmd_remote(args):
+    context = application_init(args)
+    from cola.widgets import editremotes
+    view = editremotes.new_remote_editor()
+    return application_start(context, view)
+
+
+def cmd_search(args):
+    context = application_init(args)
+    from cola.widgets.search import search
+    view = search()
+    return application_start(context, view)
+
+
+def cmd_stash(args):
+    context = application_init(args)
+    from cola.widgets.stash import stash
+    view = stash()
+    return application_start(context, view)
+
+
+def cmd_tag(args):
+    context = application_init(args)
+    from cola.widgets.createtag import new_create_tag
+    view = new_create_tag(name=args.name, ref=args.ref, sign=args.sign)
+    return application_start(context, view)
+
+
+# Windows shortcut launch features:
```

Comment:
It's not especially easy to add logic like that to the installer. It could be done, but even if it were easier, I'd advise against it, because:

- As a general design principle, I think the installer should be as stupid as possible. It's harder to test and debug installer code. If someone checks out the git-cola repo on Windows and runs it from source, what is going on should be as similar as practical to what happens when it's installed. The more the installer does, the greater the differences between a source checkout and an installed copy.
- Especially if we will want to prompt the user to manually tell us where git is, I imagine we can implement and maintain that much better using PyQt than using NSIS.
- Finding git at install time means the user needs to install git before git-cola. By finding it at runtime, they can do it in either order, so the GUI could prompt them to go and install it.

I do think we should ultimately eliminate the `--git-path` option in favour of storing the location in a file, but that's a significant enough change that I think it's better done as a separate PR.